api/src/utils/csv-utils/csv-config-validation.ts

```diff
@@ -126,105 +154,119 @@ export const validateCSVHeaders = (worksheet: WorkSheet, config: CSVConfig): CSV
- * @param {CSVRow[]} mutableRows - The mutable rows array
+ * @param {CSVCellSetter} setCellValue - The header configuration
  * @returns {*} {CSVRow[]} - The updated row
  */
-export const executeSetCellValue = (params: CSVParams, headerConfig: CSVHeaderConfig, mutableRows: CSVRow[]) => {
-  const row = { ...mutableRows[params.rowIndex] };
-
-  const headerKey = params.staticHeader?.toUpperCase() ?? params.header.toUpperCase();
-  const cellValue = headerConfig?.setCellValue?.(params) ?? params.mutateCell;
+export const executeSetCellValue = (params: CSVParams, setCellValue?: CSVCellSetter) => {
+  const header = params.staticHeader?.toUpperCase() ?? params.header.toUpperCase();
+  const cell = setCellValue?.(params) ?? params.mutateCell;
 
-  // Remove the aliased header if it is not the static header
-  if (params.staticHeader && params.header !== params.staticHeader) {
-    delete row[params.header as Uppercase<string>];
-  }
-
-  row[headerKey as Uppercase<string>] = cellValue;
-
-  mutableRows[params.rowIndex] = row;
+  return { header, cell };
 };
 
 /**
  * Execute the CSVConfig `validateCell` callback for the cell.
  *
- * Note: This mutates the CSV errors array `mutableErrors`.
- *
  * @param {CSVParams} params - The CSV parameters
- * @param {CSVHeaderConfig} headerConfig - The header configuration
- * @param {CSVError[]} mutableErrors - The mutable errors array
- * @returns {*} {void}
+ * @param {CSVCellValidator} validateCell - The cell validator
+ * @returns {*} {CSVError[]}
  */
-export const executeValidateCell = (
-  params: CSVParams,
-  headerConfig: CSVHeaderConfig,
-  mutableErrors: CSVError[]
-): void => {
-  if (!headerConfig.validateCell) {
-    return;
-  }
-
-  const cellErrors = headerConfig.validateCell(params);
-
-  if (cellErrors.length) {
-    cellErrors.forEach((error) => {
-      mutableErrors.push({
-        error: error.error,
-        solution: error.solution,
-        values: error.values ?? null,
-        cell: (error.cell === undefined ? params.cell : error.cell) ?? null, // Use cell value if intentionally null
-        header: (error.header === undefined ? params.header : error.header) ?? null, // Use header value if intentionally null
-        // WorksheetRowIndexSymbol is the original row index from the worksheet ie: before filtering empty rows
-        row: error.row ?? params.row[WorksheetRowIndexSymbol] + 1 ?? params.rowIndex + 2 // headers: 1, data row: 2
-      });
-    });
-  }
+export const executeValidateCell = (params: CSVParams, validateCell: CSVCellValidator): CSVError[] => {
+  const cellErrors = validateCell(params);
+
+  return cellErrors.map((error) => ({
+    error: error.error,
+    solution: error.solution,
+    values: error.values ?? null,
+    cell: (error.cell === undefined ? params.cell : error.cell) ?? null, // Use cell value if intentionally null
+    header: (error.header === undefined ? params.header : error.header) ?? null, // Use header value if intentionally null
```

Comment:
nit: id be interested in knowing more about this code, why the values might be intentionally null, etc.

Reply:
Maybe add more detail to these comments, for future maintainers. Unless the reason would be obvious to someone that read through the code in more detail (rather than bouncing around like I am).